.github/workflows/progress-tracker.yml

```diff
@@ -0,0 +1,80 @@
+name: Progress Tracker
+
+on:
+  workflow_dispatch:
+  issues:
+    types:
+      - edited
+
+jobs:
+  track-progress:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/github-script@v2
+        with:
+          script: |
+            // Update the following two constants each year
+            const FILTER_LABEL = '2020 chapter'
+            const TRACKER_ISSUE_ID = 989
+
+            const parseTasks = text => {
+              const tasks = []
+              text.split('\n').forEach(line => {
+                if (/^(\s{2,})?([*+-]|\d+\.)\s+\[ \]\s+\S+/.test(line)) {
+                  tasks.push('')
+                }
+                if (/^(\s{2,})?([*+-]|\d+\.)\s+\[[xX]\]\s+\S+/.test(line)) {
+                  tasks.push('☑')
+                }
+              })
+              return tasks
+            }
+
+            const parseLinks = text => {
+              const links = {}
+              text.split('\n').forEach(line => {
+                const linkMap = {
+                  draft: /^\[~draft-doc\]:\s*(?<link>\S+)/,
+                  sql: /^\[~sql-dir\]:\s*(?<link>\S+)/,
+                  results: /^\[~results-sheet\]:\s*(?<link>\S+)/
+                }
+                for (const [k, v] of Object.entries(linkMap)) {
+                  const m = line.match(v)
+                  if (m) {
+                    links[k] = m.groups.link
+                  }
+                }
+              })
+              return links
+            }
+
+            const taskIssues = await github.paginate(github.issues.listForRepo, {
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              direction: 'asc',
+              labels: FILTER_LABEL
+            })
+
+            let report = ''
+            for (const issue of taskIssues) {
+              const tasksStatus = parseTasks(issue.body)
+              const links = parseLinks(issue.body)
+              if(!report) {
+                report = `Milestones | ${tasksStatus.map((v, i) => i + 1).join(' | ')}\n`
+                report += `:----------|:${tasksStatus.map(() => '-').join(':|:')}:\n`
+              }
+              report += `[${issue.title.replace(/\s+\d+$/, '')}](${issue.html_url}) ${links.draft && links.draft != '#' ? `• [🗎](${links.draft})` : ''} | ${tasksStatus.join(' | ')}\n`
+            }
+
+            const progressIssue = await github.issues.get({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              issue_number: TRACKER_ISSUE_ID
+            })
+
+            github.issues.update({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              issue_number: TRACKER_ISSUE_ID,
+              body: progressIssue.data.body.replace(/<!-- REPORT START -->[\s\S]*<!-- REPORT END -->/, `<!-- REPORT START -->\n${report}<!-- REPORT END -->`)
```

Comment:
Does this do an update even if nothing's changed? Should we check first or simpler just to do the update and let GitHub worry about it?

Reply:
I was going to check for changes before updating the issue, but my early tests suggested that GitHub ignores the update if the content is same, so it was much simpler to leave it for GH to handle.